runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java

```diff
@@ -290,11 +290,11 @@ public void setTokenSource(TokenSource tokenSource) {
      */
     public List<Token> getTokens(int start, int stop, Set<Integer> types) {
         lazyInit();
-		if ( start<0 || stop>=tokens.size() ||
-			 stop<0  || start>=tokens.size() )
+		if ( start<0 || stop>=size() ||
+			 stop<0  || start>=size() )
```

Comment:
:exclamation: These are currently bounds checks for direct access to `tokens`. (See previous comments for details.)